### DISCUSSION TITLE
Would adding `jay-babu/mason-null-ls.nvim` be a good idea ?

### DISCUSSION THREAD

**Discussion by loiccoyle** [LazyVim/LazyVim#390] [Q&A] [answered]
Currently lsp severs are installed auto-magically using `williamboman/mason-lspconfig.nvim`.

It seems that `jay-babu/mason-null-ls.nvim`  does the same but for `null-ls` sources. Maybe it makes sense to add the same machinery to auto-magically install the `null-ls` sources as well ?

From reading the README, it seems like you would only need to add something like:
```lua
require("mason-null-ls").setup({
    ensure_installed = nil,
    automatic_installation = true,
    automatic_setup = false,
})
```
I haven't tried it though.

**Comment by dpetka2001** [comment, marked as answer]
My understanding is that most null-ls sources are part of `mason-registry`. And as I look in `source.lua` most mappings seem to have the same names as mason package names. Will that be really useful? You might just as well install them via `mason` or am I missing something else?

**Reply by loiccoyle** [reply]
I don't know what `source.lua` is, but yes sure you can install them manualy through `mason`, the same is true for lsp servers and yet they are installed through `mason-lspconfig`. I think the real benefit would be consistency between the `null-ls` and the `lspconfig` spec. And not having to edit both the `null-ls` spec and the `mason` when adding a `null-ls` source.

But I agree this very much is quality of life.

**Reply by dpetka2001** [reply]
You can install formatters and linters automatically with mason too and then you have to enable the coresponding sources in null-ls. No need for anything manual. If there were more inconsistensies for the mappings between null-ls sources and mason package names it would be one thing. But many of the mappings have the same names, so I don't see what more it could provide. But of course this is only my opinion and we should wait for the maintainer's as well.

**Reply by loiccoyle** [reply]
Thanks for the input, I'm aware that you can install them with `mason` automatically, as I said, the real benefit would be consistency between the `null-ls` and the `lspconfig` spec. And not having to edit both the `null-ls` spec and the `mason` spec when adding a `null-ls` source

**Reply by dpetka2001** [reply]
Ok i understand better now the benefit of configuring only one spec instead of 2. Thnx for pointing that out. I didn't read the readme carefully enough.

**Reply by folke** [reply]
I don't think this should be a LazyVim default. You're free to add it to your own config of course
